compiler/noirc_evaluator/src/ssa/opt/assert_constant.rs

```diff
@@ -68,14 +97,27 @@ impl Function {
 fn check_instruction(
     function: &mut Function,
     instruction: InstructionId,
+    inside_empty_loop: bool,
 ) -> Result<bool, RuntimeError> {
-    let assert_constant_id = function.dfg.import_intrinsic(Intrinsic::AssertConstant);
-    let static_assert_id = function.dfg.import_intrinsic(Intrinsic::StaticAssert);
+    let assert_constant_id = function.dfg.get_intrinsic(Intrinsic::AssertConstant);
+    let static_assert_id = function.dfg.get_intrinsic(Intrinsic::StaticAssert);
+    if assert_constant_id.is_none() && static_assert_id.is_none() {
+        return Ok(true);
+    }
+
     match &function.dfg[instruction] {
         Instruction::Call { func, arguments } => {
-            if *func == assert_constant_id {
+            let is_assert_constant = Some(*func) == assert_constant_id.copied();
+            let is_static_assert = Some(*func) == static_assert_id.copied();
+
+            // Skip known assertions inside empty loops
```

Comment:
```suggestion
            // Skip assertions inside known empty loops
```

This sentence seems weird. Is this what you meant?

Reply:
Yeah, that is what I meant.